raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java

```diff
@@ -159,23 +160,26 @@ public static Path markForDelete(Path logDir, OffsetAndEpoch snapshotId) {
         }
     }
 
-    public static long lastContainedLogTimestamp(RawSnapshotReader reader) {
-        try (RecordsSnapshotReader<ByteBuffer> recordsSnapshotReader =
-             RecordsSnapshotReader.of(
-                 reader,
-                 IdentitySerde.INSTANCE,
-                 new BufferSupplier.GrowableBufferSupplier(),
-                 KafkaRaftClient.MAX_BATCH_SIZE_BYTES,
-                 true
-             )
-        ) {
-            return recordsSnapshotReader.lastContainedLogTimestamp();
+    public static long lastContainedLogTimestamp(RawSnapshotReader reader, LogContext logContext) {
+        try (var bufferSupplier = new BufferSupplier.GrowableBufferSupplier()) {
+            try (RecordsSnapshotReader<ByteBuffer> recordsSnapshotReader =
+                RecordsSnapshotReader.of(
+                    reader,
+                    IdentitySerde.INSTANCE,
+                    bufferSupplier,
+                    KafkaRaftClient.MAX_BATCH_SIZE_BYTES,
+                    true,
+                    logContext
+                )
+            ) {
+                return recordsSnapshotReader.lastContainedLogTimestamp();
+            }
         }
```

Comment:
How about using a single try block for this?
```suggestion
        try (var bufferSupplier = new BufferSupplier.GrowableBufferSupplier();
             RecordsSnapshotReader<ByteBuffer> recordsSnapshotReader = RecordsSnapshotReader.of(
                 reader,
                 IdentitySerde.INSTANCE,
                 bufferSupplier,
                 KafkaRaftClient.MAX_BATCH_SIZE_BYTES,
                 true,
                 logContext
             )) {
            return recordsSnapshotReader.lastContainedLogTimestamp();
        }
```

Reply:
Done. Fixed.